### DISCUSSION TITLE
Decorators dependencies

### DISCUSSION THREAD

**Discussion by AlexanderTserkovniy** [fastify/fastify#2959] [Q&A]
Hi guys,

Why do we need to pass dependencies into decorators?

As I know we can reach each one directly via `this`.
If it is some kind of dependency injection, how then I can reach exact dependency?

**Comment by mcollina** [comment]
I don't really understand the question, could you clarify it a bit? I usually have a plugin calling `decoreateRequest('something')` and another accessing it via `this.something`. This works as a some very crude dependency injection.

**Reply by AlexanderTserkovniy** [reply]
The very first statement of decorators says:

### Usage

**decorate(name, value, [dependencies])**

So my questions is, why would I need some `[dependencies]` if I can access them just through the `context`.

**Reply by mcollina** [reply]
I even forget we had this.
Apparently this goes back to 2017 and v0.13.0. added by @delvedor in https://github.com/fastify/fastify/commit/d516ccfce7d8b2977052707d6f798522af961b03

@delvedor do you remember what was the reasoning behind this?

Anyway, I think we could safely delete this feature.

**Reply by AlexanderTserkovniy** [reply]
Thank you very much!

**Reply by StarpTech** [reply]
https://github.com/fastify/fastify/blob/main/docs/Decorators.md#usage ?

> The dependencies parameter is an optional list of decorators that the decorator being defined relies upon. This list is simply a list of string names of other decorators. In the following example, the "utility" decorator depends upon "greet" and "log" decorators:

> If a dependency is not satisfied, the decorate method will throw an exception. The dependency check is performed before the server instance is booted. Thus, it cannot occur during runtime.
